.travis.yml

```diff
@@ -53,6 +53,22 @@ matrix:
     - os: osx
       env: BUILD_TYPE=Release VERBOSE=1 CXX_FLAGS=-std=c++11
       if: type != pull_request
+    - os: linux
+      compiler: gcc
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-8
+      env: CXX=g++-8 && CC=gcc-8 BUILD_TYPE=Release VERBOSE=1 CXX_FLAGS=-std=c++11
```

Comment:
You probably want to also set `AR=gcc-ar-8 && RANLIB=gcc-ranlib-8`.

Reply:
@tanzislam what is the reason why I need to set `AR=gcc-ar-8 && RANLIB=gcc-ranlib-8`?

Reply:
[This StackOverflow answer](https://stackoverflow.com/a/48815662/1433768) explains that the `gcc-`-prefixed versions get shipped with the compiler rather than the OS. (We can't change the system default to GCC 8 here because [that would require sudo](https://github.com/travis-ci/travis-ci/issues/3668).)

I'm not sure exactly how the GCC-shipped ones are different from the OS-shipped ones. (It may well be that right now they only matter for using the Link Time Optimization feature, which we don't use here, and thus it may not make a difference after all.) It just seems safer to use tools from the same package/installation.

Reply:
@tanzislam Thanks. Do you know how to fix the error from `Travis`? The error shows above comment.